scripts/apply_migration.py

```
@@ -0,0 +1,371 @@
+        
+        Raises:
+            MigrationError: If connection fails or credentials are missing
+        """
+        try:
+            self.supabase_client = get_supabase_client()
+            logger.info("Successfully connected to Supabase database")
+        except ValueError as e:
+            raise MigrationError(f"Database connection failed: {e}")
+        except Exception as e:
+            raise MigrationError(f"Unexpected error connecting to database: {e}")
+    
```

Flake8 (7.2.0)

[warning] blank line contains whitespace

(W293)